### PR TITLE
Xenobiology Overhaul (WIP)

### PR DESCRIPTION
**Pull request by That0nePerson**
## About The Pull Request
Replaces the typical slimes of xenobiology with large slime mold-like blobs that spread out over the ground. Replaces the comfy xenobio console with manual work to cut, harvest, and feed the slimes. Overall goal: make xenobio faster with less waiting around for slimes to respawn, but make the xenobiologist put more work into it rather than sitting at a console. Encourage cooperation with other departments by adding rewards for giving out extracts.
More detailed design doc [here](https://docs.google.com/document/d/18hC6Jxl4kFzc95oyzefTAOgz253xXZ3k85Oii3sEn0U/edit?usp=sharing)

## Why It's Good For The Game
Xenobio is in a bit of a bad spot currently. It takes a horrible amount of time, little to no effort or skill, and gives arguably OP gear. It has many ways to help other departments, but it always turns into a fort that the xenobiologists stay in for the entire round, sitting at their consoles.

## Checklist

- [x] Basic Slimeblob functionality (hunger, aggression, spreading, eating)

- [ ] Colors and mutations (different colors have different step effects)

- [ ] Tools

- [ ] Mapping

- [ ] Slime Extract Rebalacing (currently up in the air)

- [ ] More to be added

## Changelog
:cl: St0rmC4ast3r, zeskorion, That0nePerson
add: Nanotrasen has discovered a new form of xenobiology slimes, and will now be researching them intensely. The xenobiology wing has been redesigned to accommodate these new lifeforms.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
now this looks cool

**Comment by Vexylius** [comment]
Now that's some good PR right there. Post any screens you have here, when you'll have any

**Comment by Sylphily** [comment]
Now here is my question, what happens if it gets out into the station

**Comment by Rukofamicom** [comment]
> Now here is my question, what happens if it gets out into the station

This should be relocated to science branch of the lavaland outpost, and existing Xenobio lab/sat can be repurposed on the stations.

**Comment by ItsTisme** [comment]
2 things.

1 link to the original PR. I think TG tryed to implement this too.

2 this kinda retroactivly nulls the current xenobio bountys. might wana double check if thats all good and shit.

Also ye we can move it to lavaland science post for more room to work with

**Comment by Sylphily** [comment]
> > Now here is my question, what happens if it gets out into the station
> 
> This should be relocated to science branch of the lavaland outpost, and existing Xenobio lab/sat can be repurposed on the stations. 

To be honest I think it should stay on the station, moving it to lavaland would make it even more isolated than It already is

**Comment by ItsTisme** [comment]
Black - Will inject/splash 1u of Advanced Mutation Toxin on the player
Oil - Explodes when stepped on. Blast can blow limbs off but cannot cause floor/hull breaches.
(alternatively causes the player to slip over space lube style when trodden on) 
Green - Will inject/splash 2-5u of Mutation Toxin, turning players into slime people.


Accoring to the doc, please dont do this.

**Comment by That0nePerson** [comment]
@Autisem Uhh there is no original pr, this is my and st0rm's code and the idea came from beecord. Also, yes I can talk to ruko about the extracts, I've discussed it with him before.
And everything on the doc is just a suggestion, anyone can suggest changes. Put cool ideas on there if you want, nothing is final. I'm planning on doing an open test of it when all of the basics (everything before extract rebalancing) is done

File: code/modules/mob/living/simple_animal/slime/slimeblob/xenoblob.dm [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+	plane = FLOOR_PLANE
+	icon_state = "creep"
+	max_integrity = 30
+	//canSmoothWith = list(/obj/structure/xenoblob/creep, /turf/closed/wall)
+	//smooth = SMOOTH_MORE
+
+	var/last_expand = 0 // Last world.time this creep expanded
+	var/growth_cooldown = 100
+	var/node_timer = 0
+	var/absorb_time = 200 // How long a (non-bio protected) mob has to be buckled (and dead) before becoming a node
```
Comment (Archanial) [inline code/modules/mob/living/simple_animal/slime/slimeblob/xenoblob.dm:47]: I suggest you use SECONDS instead of int values

**Comment by ghost** [comment]
> Also ye we can move it to lavaland science post for more room to work with

If the lavaland base remained in disrepair it would be a massive pain to repair it every single round just to get xenobio going.

**Comment by Sylphily** [comment]
Maybe try to redo the slime races to be more fitted to how the slime blobs work

**Pull request closed without merging**